### PR TITLE
User can sign up and log in [wip]

### PR DESCRIPTION
**Pull request by designerofthing**
- adding user identification and authentication functionality with Devise gem.
- adding bootstrap framework for css styling.
- testing and retesting with Rspec, Warden and FactoryBot.  
- Deployed branch to Heruko.

What I learnt -
-How to add Devise to a rails project
-The use of Warden and FactoryBot in feature and unit testing.
-How to add a CSS framework to a rails project
-Git flow, still not perfect but becoming more familiar.

### CODE REVIEW

File: app/views/articles/show.html.erb [hunk 1]
```diff
@@ -1,7 +1,12 @@
 <h1><%= @article.title %></h1><br />
 <p><%= @article.content %></p>
-<%= link_to "Edit Article", edit_article_path(@article) %>
-<%= link_to "Delete Article", article_path(@article),
-    method: :delete,
-    data: { confirm: 'Are you sure?'} %>
-<%= link_to 'Back', articles_path %>
\ No newline at end of file
+
+<% if user_signed_in? %>
```
Comment (faraznaeem) [inline app/views/articles/show.html.erb:4]: indentaion is off in this file

File: app/views/articles/index.html.erb [hunk 2]
```diff
@@ -1,6 +1,19 @@
-<ul>
-  <% @articles.each do |article| %>
-    <li><%= link_to article.title, article_path(article) %></li><br />
+
+<div class="jumbotron">
```
Comment (oliverochman) [inline app/views/articles/index.html.erb:2]: Please remove empty lines and indent this file properly

File: config/routes.rb [hunk 3]
```diff
@@ -3,6 +3,7 @@
   get 'articles/index'
```
Comment (faraznaeem) [inline config/routes.rb:3]: Please do not use non-resourceful routing

File: spec/models/user_spec.rb [hunk 4]
```diff
@@ -1,5 +1,25 @@
+# frozen_string_literal: true
+
 require 'rails_helper'
 
-RSpec.describe User, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
+feature 'check user db table' do
+  let(:user) { FactoryBot.create(:user, email: 'testing@gmail.com') }
```
Comment (oliverochman) [inline spec/models/user_spec.rb:6]: Why do you have this line here?

you should have it within the describe block for factory

File: spec/features/user_can_sign_up_spec.rb [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+feature 'user can sign up' do
+  before do
+    visit root_path
+    click_on 'Sign Up'
+  end
+
+  it 'user should be on the sign up page' do
+      expect(page).to have_content 'Sign up'
+  end
+
```
Comment (oliverochman) [inline spec/features/user_can_sign_up_spec.rb:14]: it would have been good to see some sad paths tested here

File: spec/models/user_spec.rb [hunk 6]
```diff
@@ -1,5 +1,25 @@
+# frozen_string_literal: true
+
 require 'rails_helper'
 
-RSpec.describe User, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
+feature 'check user db table' do
```
Comment (faraznaeem) [inline spec/models/user_spec.rb:5]: This should be a model spec, not a feature

File: app/views/articles/show.html.erb (hunk 1 above)
Comment (designerofthing) [reply app/views/articles/show.html.erb:4]: OK, sorry.

File: spec/models/user_spec.rb (hunk 6 above)
Comment (designerofthing) [reply spec/models/user_spec.rb:5]: I understand

File: config/routes.rb (hunk 3 above)
Comment (designerofthing) [reply config/routes.rb:3]: I need some more explanation on this

File: app/views/articles/index.html.erb (hunk 2 above)
Comment (designerofthing) [reply app/views/articles/index.html.erb:2]: ok

File: spec/models/user_spec.rb (hunk 4 above)
Comment (designerofthing) [reply spec/models/user_spec.rb:6]: didn't fully understand, but do now.  I have moved it

File: spec/features/user_can_sign_up_spec.rb (hunk 5 above)
Comment (designerofthing) [reply spec/features/user_can_sign_up_spec.rb:14]: sorry, i have done two now.

**Pull request merged**